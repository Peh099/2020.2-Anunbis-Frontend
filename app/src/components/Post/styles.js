import styled from 'styled-components';
import ReportIconBW from '../../assets/images/Report_Icon_BW.png';
import ReportIconC from '../../assets/images/Report_Icon_C.png';

export const PostStyle = styled.div`
  background-color: var(--yellow);
  height: 160px;
  display: flex;
  padding: 10px;
  width: 100%;
  margin: 20px 0px 0px 0px;
  box-shadow: 5px 5px 5px rgba(0, 0, 0, 50%);
  justify-content: space-evenly;
  flex-direction: column;
  border-radius: 10px;
`;

export const HeaderPost = styled.div`
  display: grid;
  grid-template-columns: min(30vw, 350px) min(20vw, 220px) 30px;
  height: fit-content;

  button {
    border: none;
    margin-inline: 5px;
    height: 20px;
    width: 20px;
    padding: 0px;
    display: flex;
    align-items: center;
    justify-content: center;
    background: url(${ReportIconBW});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 80%;
  }

  button:hover {
    background: url(${ReportIconC});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 80%;
  }

  @media (max-width: 400px) {
    grid-template-columns: min(40vw, 200px) min(25vw, 150px) 15vw;
    overflow-x: auto;
  }
`;

export const ContentPost = styled.div`
  background-color: var(--lightWhite);
  width: 100%;
  height: fit-content;
  margin-top: auto;
  padding: 10px;
  font: 11px Roboto;
  border-radius: 5px;
  overflow-y: auto;
  word-break: break-all;

  ::-webkit-scrollbar {
    width: 3px;
  }

  ::-webkit-scrollbar-thumb {
    background-color: var(--black);
  }
  @media (max-width: 700px) {
    font-size: 10px;
  }
`;

export const FeedbacksDiv = styled.div`
  display: flex;
  justify-content: space-between;

  button {
    border: none;
    height: 15px;
    width: 60px;
    margin-inline: 60px;
    background-repeat: no-repeat;
    background-position: left;
    background-size: 40%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    color: rgba(61, 58, 58, 0.603);
    box-shadow: none;
    @media (max-width: 400px) {
      height: 15px;
      width: 30px;
      margin-inline: 20px;
    }
  }
`;

export const InfoStudent = styled.div`
  font: 12px Roboto;
  padding-left: 10px;
  @media (max-width: 500px) {
    font: 11px Roboto;
    flex-wrap: wrap;
  }
  @media (max-width: 300px) {
    font: 10px Roboto;
    text-overflow: ellipsis;
    overflow: hidden;
  }
`;

export const Name = styled.div`
  color: #696969;
  @media (max-width: 700px) {
    font: 11px Roboto;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
  }
`;

export const Rating = styled.div`
  display: grid;
  grid-template-columns: min(20vw, 120px) min(15vw, 100px);
  font-size: 11px;    
  @media (max-width: 760px) {
    grid-template-columns: 80% 60%;
    label {
      font-size: 10px;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;}
  }
`;
