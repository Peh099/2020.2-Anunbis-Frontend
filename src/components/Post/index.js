import React from 'react';
import styled from 'styled-components';

const Header = ({ children }) => {
    return (
        <Header_Post>
            <Info_Student>
                [{children.discipline_code}] {children.discipline_name}<br />
                {children.name_course}<br />
                    Data: {children.date}<br />
                <Name>
                    {children.name_studant}
                </Name>
            </Info_Student>
            <Rating>
                Nota: {children.rating}
            </Rating>
        </Header_Post>
    )
}

const Content = ({ children }) => {
    return (
        <Content_Post>
            {children.content}
        </Content_Post>
    )
}

export default function Post({ children }) {
    return (
        <PostStyle>
            <Header>
                {children[0].props}
            </Header>
            <Content>
                {children[1].props}
            </Content>
        </PostStyle>
    );
}

const PostStyle = styled.div`
    background-color: #FFD54F;
    width: 350px;
    height: 100px;
    margin-bottom: 10px;
    display: flex;
    justify-content: space-evenly;
    flex-direction: column;
    border-radius: 10px;
`;

const Header_Post = styled.div`
    display: block;
    height: 40px;
    display: flex;
    justify-content: space-between;
`;

const Info_Student = styled.div`
    height: 32px;
    font: 10px Roboto;
    padding-left: 10px;
`;

const Name = styled.div`
    color: #696969;
`;

const Rating = styled.div`
    padding: 8px;
`;

const Content_Post = styled.div`
    background-color: #FFFDE7;
    width: 315px;
    height: 45px;
    margin-left: auto;
    margin-right: auto;
    font: 11px Roboto;
    padding-left: 5px;
    border-radius: 5px;
    overflow-y: auto;
    word-break: break-all;
`;