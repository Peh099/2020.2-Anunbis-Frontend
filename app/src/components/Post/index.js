import React from 'react';
import Button from '../Button';
import Agreed_Icon_BW from '../../assets/images/Agreed_Icon_BW.png';
import Agreed_Icon_C from '../../assets/images/Agreed_Icon_C.png';
import Disagreed_Icon_BW from '../../assets/images/Disagreed_Icon_BW.png';
import Disagreed_Icon_C from '../../assets/images/Disagreed_Icon_C.png';
import { PostStyle, HeaderPost, InfoStudent, Name, Rating, ContentPost, FeedbacksDiv } from './styles.js'
import api from '../../services/Api'
import Report from '../Report';
import Popup from '../Popup';
import { isProfessor, isStudent} from '../../services/Auth';

const Header = ({ post, onClickReport }) => {
    return (
        <HeaderPost>
            <InfoStudent>
                [{post.discipline.discipline_code}] {post.discipline.name}<br />
                {post.student.course.name}<br />
                    Data: {post.post_date}<br />
                <Name>
                    {post.student.name}
                </Name>
            </InfoStudent>
            <Rating>
                Nota: {post.rating}
            </Rating>
            <Button type="button" backColor="rgba(255, 0, 0, 0)" onClick={onClickReport} shadow="None"/>
        </HeaderPost>
    )
}

const Content = ({ children }) => {
    return (
        <ContentPost>
            {children}
        </ContentPost>
    )
}


const Feedbacks = ({ post, Onclick }) => {
    const isAgreed = post.feedbacks.is_agreed || isProfessor();
    const isDisagreed = post.feedbacks.is_disagreed;
    const countAgrees = post.feedbacks.agrees;
    const countDisagrees = post.feedbacks.disagrees;

    function clickAgree() {
        if(isStudent()){
            const body = { 'id_post': post.id_post }
            api.post("/post/agree", body)
                .then(res => Onclick(res.data))
        }
    }

    function clickDisagree() {
        if(isStudent()){
        const body = { 'id_post': post.id_post }
        api.post("/post/disagree", body)
            .then(res => Onclick(res.data))
        }
    }

    return (
        <FeedbacksDiv>
            <Button text={countAgrees} backImage={isAgreed ? Agreed_Icon_C : Agreed_Icon_BW} backColor="rgba(255, 0, 0, 0)" shadow="1px 1px 1px rgba(0, 0, 0, 20%)" onClick={() => clickAgree()} />
            <Button text={countDisagrees} backImage={isDisagreed ? Disagreed_Icon_C : Disagreed_Icon_BW} backColor="rgba(255, 0, 0, 0)" shadow="1px 1px 1px rgba(0, 0, 0, 20%)" onClick={() => clickDisagree()} />
        </FeedbacksDiv>
    )
}

export default function Post({ post }) {
    const [currentPost, setCurrentPost] = React.useState(post)
    const [reportBox, setReportBox] = React.useState(false)

    return (<>
        {reportBox && <Popup><Report post={currentPost} close={() => setReportBox(false)} /></Popup>}
        <PostStyle>
            <Header post={currentPost} onClickReport={() => setReportBox(true)} />
            <Content>
                {currentPost.content}
            </Content>
            <Feedbacks Onclick={setCurrentPost} post={currentPost} />
        </PostStyle>
    </>
    );
}
