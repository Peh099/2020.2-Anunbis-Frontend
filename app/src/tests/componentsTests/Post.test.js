import renderer from 'react-test-renderer';
import Post from '../../components/Post';

describe('Snapshot Post component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Post />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
