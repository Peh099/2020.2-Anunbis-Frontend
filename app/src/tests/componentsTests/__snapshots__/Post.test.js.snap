// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Post component matches the snapshot 1`] = `
<div
  className="sc-gtsrHT dDYQua"
>
  <div
    className="sc-dlnjwi jUxskP"
  >
    <div
      className="sc-jSFjdj iJMXeY"
    >
      [
      ]
      <br />
      <br />
      Data: 
      <br />
      <div
        className="sc-gKAaRy iKxJej"
      />
    </div>
    <div
      className="sc-iCoGMd jRTzpo"
    >
      <label>
        <b>
          Nota Geral:
        </b>
         
      </label>
      <label>
        <b />
      </label>
      <label>
        Didatica:
      </label>
      <label />
      <label>
        Metodologia:
      </label>
      <label />
      <label>
        Avaliações Coerentes:
      </label>
      <label />
      <label>
        Disponibilidade:
      </label>
      <label />
    </div>
    <button
      className="sc-bdnxRM bPQZrU"
      onClick={[Function]}
      type="button"
    />
  </div>
  <div
    className="sc-hKFxyN frAbNa"
  />
  <div
    className="sc-eCApnc gGWRwP"
  >
    <button
      className="sc-bdnxRM ikeTfa"
      onClick={[Function]}
    />
    <button
      className="sc-bdnxRM iINuJo"
      onClick={[Function]}
    />
  </div>
</div>
`;
